assets/src/edit-story/components/keyboardShortcutsMenu/index.js

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useRef, useState, useCallback, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useGlobalKeyDownEffect, useKeyDownEffect } from '../keyboard';
+import WithTooltip from '../tooltip';
+import Popup, { Placement } from '../popup';
+import { Keyboard as KeyboardShortcutsButton } from '../button';
+import ShortcutMenu from './shortcutMenu';
+
+function KeyboardShortcutsMenu({
```

Comment:
When opening a modal dialog window, you **must** move focus to the first focusable element in the dialog window.

However, for this modal dialog, we don't actually have a focusable element, but we also don't have a visual "close" button. So, I recommend adding a close button to the dialog and auto-focusing it when the dialog opens.

Likewise, when a modal dialog closes, you **must** return focus to the element that opened it, so the user can keep navigating from there as if they never opened the dialog at all.

I think these two things will also remove a lot of the logic you have around timeouts and `tryToggleMenu`.

Finally, note the requirement below to "trap" tabbing inside the modal dialog. If we only have one focusable element, the tabbing will simply do nothing inside the dialog.

Here's [the ARIA guide for reference](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal) and here's [an example reference](https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/dialog.html).

Reply:
Yeah, I'll switch to the react modal as you and Pascal recommended, and that should fix up all this and remove the need for the timeout logic.